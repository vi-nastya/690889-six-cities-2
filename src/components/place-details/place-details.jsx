import React from "react";
import PropTypes from "prop-types";
import Header from "../header/header.jsx";
import ReviewForm from "../review-form/review-form.jsx";
import ReviewsList from "../reviews-list/reviews-list.jsx";
import {connect} from "react-redux";
import {Operation} from "../../reducer";
import {getOfferById} from "../../selectors/selectors";

const PlaceDetails = (props) => {
  const {placeData} = props;
  if (!placeData) {
    return <h2>Loading</h2>;
  }
  return <div>
    <div style={{display: `none`}}>
      <svg xmlns="http://www.w3.org/2000/svg">
        <symbol id="icon-arrow-select" viewBox="0 0 7 4">
          <path fillRule="evenodd" clipRule="evenodd" d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"></path>
        </symbol>
        <symbol id="icon-bookmark" viewBox="0 0 17 18">
          <path d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"></path></symbol>
        <symbol id="icon-star" viewBox="0 0 13 12"><path fillRule="evenodd" clipRule="evenodd" d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"></path>
        </symbol>
      </svg>
    </div>

    <div className="page">
      <Header/>
      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {placeData.images.map((image, index) => {
                return <div className="property__image-wrapper" key={`offer-image-` + index}>
                  <img className="property__image" src={image} alt="Photo studio"/>
                </div>;
              })}
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              <div className="property__mark">
                {placeData.isPremium ? <span>Premium</span> : ``}
              </div>
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {placeData.title}
                </h1>
                <button className="property__bookmark-button button" type="button">
                  <svg className="property__bookmark-icon" width="31" height="33">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>
              <div className="property__rating rating">
                <div className="property__stars rating__stars">
                  <span style={{width: `96%`}}></span>
                  <span className="visually-hidden">Rating</span>
                </div>
                <span className="property__rating-value rating__value">{placeData.rating}</span>
              </div>
              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {placeData.type}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {placeData.bedrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {placeData.maxAdults} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{placeData.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <ul className="property__inside-list">
                  {placeData.goods.map((item, index) => {
                    return <li className="property__inside-item" key={`inside-item-` + index}>
                      {item}
                    </li>;
                  })}
                </ul>
              </div>
              <div className="property__host">
                <h2 className="property__host-title">Meet the host</h2>
                <div className="property__host-user user">
                  <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                    <img className="property__avatar user__avatar" src={placeData.host.avatarUrl} width="74" height="74" alt="Host avatar"/>
                  </div>
                  <span className="property__user-name">
                    {placeData.host.name}
                  </span>
                  {placeData.host.isPro ? <span className="property__user-status">Pro</span> : ``}
                </div>
                <div className="property__description">
                  <p className="property__text">
                    {placeData.description}
                  </p>
                </div>
              </div>
              <section className="property__reviews reviews">
                <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">1</span></h2>
                {/* <ReviewsList/>
                <ReviewForm /> */}
              </section>
            </div>
          </div>
          <section className="property__map map"></section>
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <div className="near-places__list places__list">
              <article className="near-places__card place-card">
                <div className="near-places__image-wrapper place-card__image-wrapper">
                  <a href="#">
                    <img className="place-card__image" src="img/room.jpg" width="260" height="200" alt="Place image"/>
                  </a>
                </div>
                <div className="place-card__info">
                  <div className="place-card__price-wrapper">
                    <div className="place-card__price">
                      <b className="place-card__price-value">&euro;80</b>
                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                    </div>
                    <button className="place-card__bookmark-button place-card__bookmark-button--active button" type="button">
                      <svg className="place-card__bookmark-icon" width="18" height="19">
                        <use xlinkHref="#icon-bookmark"></use>
                      </svg>
                      <span className="visually-hidden">In bookmarks</span>
                    </button>
                  </div>
                  <div className="place-card__rating rating">
                    <div className="place-card__stars rating__stars">
                      <span style={{width: `80%`}}></span>
                      <span className="visually-hidden">Rating</span>
                    </div>
                  </div>
                  <h2 className="place-card__name">
                    <a href="#">Wood and stone place</a>
                  </h2>
                  <p className="place-card__type">Private room</p>
                </div>
              </article>

              <article className="near-places__card place-card">
                <div className="near-places__image-wrapper place-card__image-wrapper">
                  <a href="#">
                    <img className="place-card__image" src="img/apartment-02.jpg" width="260" height="200" alt="Place image"/>
                  </a>
                </div>
                <div className="place-card__info">
                  <div className="place-card__price-wrapper">
                    <div className="place-card__price">
                      <b className="place-card__price-value">&euro;132</b>
                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                    </div>
                    <button className="place-card__bookmark-button button" type="button">
                      <svg className="place-card__bookmark-icon" width="18" height="19">
                        <use xlinkHref="#icon-bookmark"></use>
                      </svg>
                      <span className="visually-hidden">To bookmarks</span>
                    </button>
                  </div>
                  <div className="place-card__rating rating">
                    <div className="place-card__stars rating__stars">
                      <span style={{width: `80%`}}></span>
                      <span className="visually-hidden">Rating</span>
                    </div>
                  </div>
                  <h2 className="place-card__name">
                    <a href="#">Canal View Prinsengracht</a>
                  </h2>
                  <p className="place-card__type">Apartment</p>
                </div>
              </article>

              <article className="near-places__card place-card">
                <div className="near-places__image-wrapper place-card__image-wrapper">
                  <a href="#">
                    <img className="place-card__image" src="img/apartment-03.jpg" width="260" height="200" alt="Place image"/>
                  </a>
                </div>
                <div className="place-card__info">
                  <div className="place-card__price-wrapper">
                    <div className="place-card__price">
                      <b className="place-card__price-value">&euro;180</b>
                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                    </div>
                    <button className="place-card__bookmark-button button" type="button">
                      <svg className="place-card__bookmark-icon" width="18" height="19">
                        <use xlinkHref="#icon-bookmark"></use>
                      </svg>
                      <span className="visually-hidden">To bookmarks</span>
                    </button>
                  </div>
                  <div className="place-card__rating rating">
                    <div className="place-card__stars rating__stars">
                      <span style={{width: `100%`}}></span>
                      <span className="visually-hidden">Rating</span>
                    </div>
                  </div>
                  <h2 className="place-card__name">
                    <a href="#">Nice, cozy, warm big bed apartment</a>
                  </h2>
                  <p className="place-card__type">Apartment</p>
                </div>
              </article>
            </div>
          </section>
        </div>
      </main>
    </div>
  </div>;
};

PlaceDetails.propTypes = {
  placeData: PropTypes.object.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    placeData: getOfferById(parseInt(ownProps.match.params.id, 10))(state),
  });
};

const mapDispatchToProps = (dispatch) => ({
  changeFavoriteHandler: (placeData) => {
    dispatch(Operation.updateFavoriteStatus(placeData));
  }
});

export {PlaceDetails};

export default connect(mapStateToProps, mapDispatchToProps)(PlaceDetails);
