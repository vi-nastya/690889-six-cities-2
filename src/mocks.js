export const MOCK_CITIES = [{
  name: `City1`,
  location: {
    latitude: 52.0,
    longitude: 4.5,
    zoom: 10
  }
},
{
  name: `City2`,
  location: {
    latitude: 20.4,
    longitude: 10.1,
    zoom: 3
  }
}
];

export const MOCK_USER = {
  id: 4,
  isPro: false,
  name: `Max`,
  avatarUrl: `img/1.png`
};

export const MOCK_OFFERS = [{
  id: 1,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
},
{
  id: 2,
  city: MOCK_CITIES[1],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 3.5,
  type: `apartment`,
  bedrooms: 1,
  maxAdults: 4,
  price: 1,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52,
    longitude: 30,
    zoom: 8
  }
},
{
  id: 3,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: true,
  rating: 2,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
}
];

export const MOCK_OFFER_UPDATED_FAVORITE = {
  id: 1,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
};

export const MOCK_OFFERS_UPDATED_FAVORITE = [{
  id: 1,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
},
{
  id: 2,
  city: MOCK_CITIES[1],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: true,
  isPremium: false,
  rating: 3.5,
  type: `apartment`,
  bedrooms: 1,
  maxAdults: 4,
  price: 1,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52,
    longitude: 30,
    zoom: 8
  }
},
{
  id: 3,
  city: MOCK_CITIES[0],
  previewImage: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: true,
  rating: 2,
  type: `apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
}
];


export const MOCK_REVIEWS = [{
  id: 1,
  user: MOCK_USER,
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
}];

export const MOCK_AUTH = {
  id: 4,
  isPro: false,
  name: `Max`,
  email: `max@test.com`,
  avatarUrl: `img/1.png`
};


/* eslint-disable camelcase */
export const MOCK_AUTH_SERVER = {
  id: 4,
  is_pro: false,
  name: `Max`,
  email: `max@test.com`,
  avatar_url: `img/1.png`
};


export const MOCK_USER_SERVER = {
  id: 4,
  is_pro: false,
  name: `Max`,
  avatar_url: `img/1.png`
};

export const MOCK_REVIEWS_SERVER = [{
  id: 1,
  user: MOCK_USER_SERVER,
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
}];


export const MOCK_OFFERS_SERVER = [{
  id: 1,
  city: MOCK_CITIES[0],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: false,
  is_premium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  max_adults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
},
{
  id: 2,
  city: MOCK_CITIES[1],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: true,
  is_premium: false,
  rating: 3.5,
  type: `apartment`,
  bedrooms: 1,
  max_adults: 4,
  price: 1,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52,
    longitude: 30,
    zoom: 8
  }
},
{
  id: 3,
  city: MOCK_CITIES[0],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: false,
  is_premium: true,
  rating: 2,
  type: `apartment`,
  bedrooms: 3,
  max_adults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
}
];

export const MOCK_OFFER_UPDATED_FAVORITE_SERVER = {
  id: 1,
  city: MOCK_CITIES[0],
  preview_image: `img/1.png`,
  images: [`img/1.png`, `img/2.png`],
  title: `Beautiful & luxurious studio at great location`,
  is_favorite: true,
  is_premium: false,
  rating: 4.8,
  type: `apartment`,
  bedrooms: 3,
  max_adults: 4,
  price: 120,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: MOCK_USER_SERVER,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 56,
    longitude: 3,
    zoom: 8
  }
};
/* eslint-enable */
