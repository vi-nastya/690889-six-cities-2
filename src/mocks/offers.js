export const offers = [
  {
    id: 0,
    title: `Wood and stone place`,
    price: 80,
    city: {name: `Amsterdam`, location: {latitude: 52.3909553943508, longitude: 4.929309666406198}},
    location: {latitude: 52.3909553943508, longitude: 4.85309666406198},
    priceText: `night`,
    type: `Private room`,
    picture: `img/apartment-01.jpg`,
    isPremium: false,
    isFavorite: true,
    images: [`img/1.png`, `img/2.png`],
    bedrooms: 2,
    rating: 4,
    maxAdults: 2,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      is–óro: true,
      name: `Angelina`,
      avatarUrl: `img/1.png`
    },
  },
  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    price: 120,
    city: {name: `Amsterdam`, location: {latitude: 52.3909553943508, longitude: 4.929309666406198}},
    location: {latitude: 52.369553943508, longitude: 4.85309666406198},
    priceText: `night`,
    type: `Apartment`,
    picture: `img/apartment-02.jpg`,
    isPremium: true,
    isFavorite: false,
    rating: 3.6,
    images: [``]
  },
  {
    id: 2,
    title: `Nice, cozy, warm big bed apartment`,
    price: 180,
    city: {name: `Amsterdam`, location: {latitude: 52.3909553943508, longitude: 4.929309666406198}},
    location: {latitude: 52.3909553943508, longitude: 4.929309666406198},
    priceText: `night`,
    type: `Apartment`,
    picture: `img/apartment-03.jpg`,
    isPremium: true,
    isFavorite: true,
    images: [``],
    rating: 2,
  },
  {
    id: 3,
    title: `Wood and stone place`,
    price: 90,
    city: {name: `Moscow`, location: {latitude: 52.3809553943508, longitude: 4.939309666406198}},
    location: {latitude: 52.3809553943508, longitude: 4.939309666406198},
    priceText: `night`,
    type: `Private room`,
    picture: `https://picsum.photos/260/200`,
    isPremium: false,
    isFavorite: false,
    images: [``],
    rating: 4.8,
  },
];
