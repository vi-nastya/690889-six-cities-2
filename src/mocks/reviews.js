export const reviews = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatarUrl: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 1,
      isPro: true,
      name: `Mark`,
      avatarUrl: `img/2.png`
    },
    rating: 2,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-10T14:13:56.569Z`
  }
];
