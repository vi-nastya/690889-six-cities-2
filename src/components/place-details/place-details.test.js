import React from "react";
import renderer from "react-test-renderer";
import {PlaceDetails} from "./place-details";
import {MOCK_OFFERS, MOCK_REVIEWS} from "../../mocks";

jest.mock(`../header/header`, () => jest.fn().mockReturnValue(null));
jest.mock(`react-router-dom`, () => ({
  Link: () => null
}));
jest.mock(`../map/map`, () => `Map`);

it(`PlaceDetails component is rendered correctly after relaunch`, () => {
  const tree = renderer.create(<PlaceDetails
    placeData={MOCK_OFFERS[0]}
    changeFavoriteHandler={jest.fn()}
    onLoadComments={jest.fn()}
    match={{params: {id: `1`}}}
    reviews={MOCK_REVIEWS}
    nearbyPlaces={[MOCK_OFFERS[1]]}
    onFormSubmit={jest.fn()}
    isSendingReview={false}
    reviewSendingError={false}
    isAuthorizationRequired={true}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
