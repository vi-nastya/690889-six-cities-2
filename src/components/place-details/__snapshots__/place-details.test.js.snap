// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceDetails component is rendered correctly after relaunch 1`] = `
<div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clipRule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clipRule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>
  <div
    className="page"
  >
    <main
      className="page__main page__main--property"
    >
      <section
        className="property"
      >
        <div
          className="property__gallery-container container"
        >
          <div
            className="property__gallery"
          >
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/1.png"
              />
            </div>
            <div
              className="property__image-wrapper"
            >
              <img
                alt="Photo studio"
                className="property__image"
                src="img/2.png"
              />
            </div>
          </div>
        </div>
        <div
          className="property__container container"
        >
          <div
            className="property__wrapper"
          >
            <div
              className="property__mark"
            >
              
            </div>
            <div
              className="property__name-wrapper"
            >
              <h1
                className="property__name"
              >
                Beautiful & luxurious studio at great location
              </h1>
              <button
                className="property__bookmark-button button "
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="property__bookmark-icon"
                  height="33"
                  width="31"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="property__rating rating"
            >
              <div
                className="property__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                className="property__rating-value rating__value"
              >
                4.8
              </span>
            </div>
            <ul
              className="property__features"
            >
              <li
                className="property__feature property__feature--entire"
              >
                apartment
              </li>
              <li
                className="property__feature property__feature--bedrooms"
              >
                3
                 Bedrooms
              </li>
              <li
                className="property__feature property__feature--adults"
              >
                Max 
                4
                 adults
              </li>
            </ul>
            <div
              className="property__price"
            >
              <b
                className="property__price-value"
              >
                €
                120
              </b>
              <span
                className="property__price-text"
              >
                 night
              </span>
            </div>
            <div
              className="property__inside"
            >
              <h2
                className="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                className="property__inside-list"
              >
                <li
                  className="property__inside-item"
                >
                  Heating
                </li>
                <li
                  className="property__inside-item"
                >
                  Kitchen
                </li>
                <li
                  className="property__inside-item"
                >
                  Cable TV
                </li>
                <li
                  className="property__inside-item"
                >
                  Washing machine
                </li>
                <li
                  className="property__inside-item"
                >
                  Coffee machine
                </li>
                <li
                  className="property__inside-item"
                >
                  Dishwasher
                </li>
              </ul>
            </div>
            <div
              className="property__host"
            >
              <h2
                className="property__host-title"
              >
                Meet the host
              </h2>
              <div
                className="property__host-user user"
              >
                <div
                  className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
                >
                  <img
                    alt="Host avatar"
                    className="property__avatar user__avatar"
                    height="74"
                    src="/img/1.png"
                    width="74"
                  />
                </div>
                <span
                  className="property__user-name"
                >
                  Max
                </span>
                
              </div>
              <div
                className="property__description"
              >
                <p
                  className="property__text"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                </p>
              </div>
            </div>
            <section
              className="property__reviews reviews"
            >
              <h2
                className="reviews__title"
              >
                Reviews · 
                <span
                  className="reviews__amount"
                >
                  1
                </span>
              </h2>
              <ul
                className="reviews__list"
              >
                <li
                  className="reviews__item"
                >
                  <div
                    className="reviews__user user"
                  >
                    <div
                      className="reviews__avatar-wrapper user__avatar-wrapper"
                    >
                      <img
                        alt="Reviews avatar"
                        className="reviews__avatar user__avatar"
                        height="54"
                        src="img/1.png"
                        width="54"
                      />
                    </div>
                    <span
                      className="reviews__user-name"
                    >
                      Max
                    </span>
                  </div>
                  <div
                    className="reviews__info"
                  >
                    <div
                      className="reviews__rating rating"
                    >
                      <div
                        className="reviews__stars rating__stars"
                      >
                        <span
                          style={
                            Object {
                              "width": "80%",
                            }
                          }
                        />
                        <span
                          className="visually-hidden"
                        >
                          Rating
                        </span>
                      </div>
                    </div>
                    <p
                      className="reviews__text"
                    >
                      A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                    </p>
                    <time
                      className="reviews__time"
                      dateTime="2019-04-24"
                    >
                      May 2019
                    </time>
                  </div>
                </li>
              </ul>
            </section>
          </div>
        </div>
        <section
          className="property__map map"
        >
          <Map
            activePoint={0}
            city={
              Array [
                52,
                4.5,
              ]
            }
            points={
              Array [
                Array [
                  56,
                  3,
                ],
                Array [
                  52,
                  30,
                ],
              ]
            }
          />
        </section>
      </section>
      <div
        className="container"
      >
        <section
          className="near-places places"
        >
          <h2
            className="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            className="near-places__list places__list"
          >
            <article
              className="near-places__card place-card"
              onMouseEnter={null}
              onMouseLeave={null}
            >
              <div
                className="place-card__mark"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="near-places__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/1.png"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      1
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button place-card__bookmark-button--active"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                />
                <p
                  className="place-card__type"
                >
                  apartment
                </p>
              </div>
            </article>
          </div>
        </section>
      </div>
    </main>
  </div>
</div>
`;
